test/e2e/nodepool_additionalTrustBundlePropagation.go

```diff
@@ -0,0 +1,110 @@
+	t.Log("Starting AdditionalTrustBundlePropagationTest.")
+}
+
+func (k *AdditionalTrustBundlePropagationTest) BuildNodePoolManifest(defaultNodepool hyperv1.NodePool) (*hyperv1.NodePool, error) {
+	nodePool := &hyperv1.NodePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      k.hostedCluster.Name + "-" + "test-additional-trust-bundle-propagation",
+			Namespace: k.hostedCluster.Namespace,
+		},
+	}
+	defaultNodepool.Spec.DeepCopyInto(&nodePool.Spec)
+	nodePool.Spec.Replicas = &oneReplicas
+
+	return nodePool, nil
+}
+
+func (k *AdditionalTrustBundlePropagationTest) Run(t *testing.T, nodePool hyperv1.NodePool, nodes []corev1.Node) {
+	t.Run("AdditionalTrustBundlePropagationTest", func(t *testing.T) {
+		e2eutil.AtLeast(t, e2eutil.Version418)
+
+		additionalTrustBundle := &corev1.ConfigMap{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "additional-trust-bundle",
+				Namespace: k.hostedCluster.Namespace,
+			},
+			Data: map[string]string{
+				"ca-bundle.crt": "dummy",
+			},
+		}
+
+		if err := k.mgmtClient.Create(k.ctx, additionalTrustBundle); err != nil {
+			t.Fatalf("failed to create additional trust bundle configmap: %v", err)
+		}
+		if err := k.mgmtClient.Get(k.ctx, crclient.ObjectKeyFromObject(k.hostedCluster), k.hostedCluster); err != nil {
+			t.Fatalf("failed to get hostedcluster")
+		}
+		k.hostedCluster.Spec.AdditionalTrustBundle = &corev1.LocalObjectReference{Name: additionalTrustBundle.Name}
+		if err := k.mgmtClient.Update(k.ctx, k.hostedCluster); err != nil {
+			t.Fatalf("failed to update hostedcluster with additional trust bundle: %v", err)
+		}
+
+		e2eutil.EventuallyObject(t, k.ctx, fmt.Sprintf("Waiting for NodePool %s/%s to begin updating", nodePool.Namespace, nodePool.Name),
+			func(ctx context.Context) (*hyperv1.NodePool, error) {
+				err := k.mgmtClient.Get(ctx, crclient.ObjectKeyFromObject(&nodePool), &nodePool)
+				return &nodePool, err
+			},
+			[]e2eutil.Predicate[*hyperv1.NodePool]{
+				e2eutil.ConditionPredicate[*hyperv1.NodePool](e2eutil.Condition{
+					Type:   hyperv1.NodePoolUpdatingConfigConditionType,
+					Status: metav1.ConditionTrue,
+				}),
+			},
+			e2eutil.WithTimeout(5*time.Minute),
+		)
+
+		e2eutil.EventuallyObject(t, k.ctx, fmt.Sprintf("Waiting for NodePool %s/%s to stop updating", nodePool.Namespace, nodePool.Name),
+			func(ctx context.Context) (*hyperv1.NodePool, error) {
+				err := k.mgmtClient.Get(ctx, crclient.ObjectKeyFromObject(&nodePool), &nodePool)
+				return &nodePool, err
+			},
+			[]e2eutil.Predicate[*hyperv1.NodePool]{
+				e2eutil.ConditionPredicate[*hyperv1.NodePool](e2eutil.Condition{
+					Type:   hyperv1.NodePoolUpdatingConfigConditionType,
+					Status: metav1.ConditionFalse,
+				}),
+				e2eutil.ConditionPredicate[*hyperv1.NodePool](e2eutil.Condition{
+					Type:   hyperv1.NodePoolAllNodesHealthyConditionType,
+					Status: metav1.ConditionTrue,
+				}),
+			},
+			e2eutil.WithTimeout(5*time.Minute),
```

Comment:
we need to increase this timeout, the test would have passed as the condition went false eventually https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_hypershift/4775/pull-ci-openshift-hypershift-main-e2e-aws/1839077602413776896/artifacts/e2e-aws/hypershift-aws-run-e2e-external/artifacts/TestNodePool_HostedCluster0/namespaces/e2e-clusters-n7lbh/hypershift.openshift.io/nodepools/example-vm9t7-test-additional-trust-bundle-propagation.yaml
We use e2eutil.WithTimeout(30*time.Minute), in the other rollout test